src/platform/unix/mod.rs

```diff
@@ -655,16 +653,14 @@ struct BackingStore {
 
 impl BackingStore {
     pub fn new(length: usize) -> BackingStore {
-        unsafe {
-            let string = CString::new(TEMP_FILE_TEMPLATE).unwrap();
-            let string_buffer = strdup(string.as_ptr());
-            let fd = mkstemp(string_buffer);
-            assert!(fd >= 0);
-            assert!(maybe_unlink(string_buffer) == 0);
-            libc::free(string_buffer as *mut c_void);
-            assert!(libc::ftruncate(fd, length as off_t) == 0);
-            Self::from_fd(fd)
-        }
+        let count = GLOBAL_SHM_COUNT.fetch_add(1, Ordering::SeqCst);
+        let timestamp = UNIX_EPOCH.elapsed().unwrap();
+        let pid = unsafe { libc::getpid() };
+        let name = CString::new(format!("/ipc-channel-shared-memory.{}.{}.{}",
+                                        count, pid,
+                                        timestamp.as_secs())).unwrap();
```

Comment:
I'm really uncertain myself whether a random hash wouldn't be preferable. I don't know whether other methods of generating random numbers -- such as `weak_rng()`, or perhaps even `StdRng::new()` in a `lazy_static` -- would avoid the FD use problem. If so, it *might* be preferable, since it's probably a tick simpler all in all... Really not sure though. I don't see a strong motivation to change it at this point.